assets/app/view/game/buy_corporation.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'view/game/actionable'
+require 'view/game/corporation'
+
+module View
+  module Game
+    class BuyCorporation < Snabberb::Component
+      include Actionable
+      needs :selected_corporation, default: nil, store: true
+      needs :show_other_players, default: nil, store: true
+
+      def render
+        @step = @game.active_step
+        @corporation = @game.current_entity
+        children = []
+
+        hidden_corps = false
+        @show_other_players = true if @step.show_other_players
+        @game.corporations.select { |item| @step.can_buy?(@corporation, item) }.each do |target|
+          if @show_other_players || target.owner == @corporation.owner || !target.owner
+            children << h(Corporation, corporation: target, selected_corporation: @selected_corporation)
+            children << render_input if target == @selected_corporation
+          else
+            hidden_corps = true
+          end
+        end
+
+        if hidden_corps
+          children << h('button',
+                        { on: { click: -> { store(:show_other_players, true) } } },
+                        'Show corporations from other players')
+        elsif @show_other_players
+          children << h('button',
+                        { on: { click: -> { store(:show_other_players, false) } } },
+                        'Hide corporations from other players')
+        end
+
+        h(:div, children)
+      end
+
+      def render_input
+        max_price = (@selected_corporation.share_price.price * 1.5).ceil
+        min_price = (@selected_corporation.share_price.price * 0.5).ceil
+        input = h(:input, style: { marginRight: '1rem' }, props: {
+                    value: @selected_corporation.share_price.price,
+                    type: 'number',
+                    min: min_price,
+                    max: max_price,
+                    size: max_price,
+                  })
+
+        buy_click = lambda do
+          price = input.JS['elm'].JS['value'].to_i
+          buy = lambda do
+            process_action(Engine::Action::BuyCorporation.new(
+              @corporation,
+              corporation: @selected_corporation,
+              price: price,
+            ))
+            store(:selected_corporation, nil, skip: true)
+          end
+
+          if @selected_corporation.owner == @corporation.owner
```

Comment:
this stuff looks copy and pasted from the company, can you instead reuse the buy companies view

Reply:
Yeah it is. So should I create a new component thats referenced by both?

Reply:
try to get buy_company to with corporations

Reply:
I think it is not really the best idea to mix the buy company and buy corporation view. There are some differences and in the end it would just be a if(corporation) render_buy_corporation else render_buy_companies. Then it is clearer to have 2 seperate classes. 

I grouped the input into an own component, which shares nearly all the code except the input value parameters.